backends/memory/memory_backend.go

```diff
@@ -7,69 +7,107 @@ import (
 	"sync"
 	"time"
 
+	"github.com/acaloiaro/neoq/config"
+	nctx "github.com/acaloiaro/neoq/context"
+	"github.com/acaloiaro/neoq/handler"
+	"github.com/acaloiaro/neoq/internal"
+	"github.com/acaloiaro/neoq/jobs"
+	"github.com/acaloiaro/neoq/logging"
+	"github.com/acaloiaro/neoq/types"
 	"github.com/guregu/null"
-	"github.com/iancoleman/strcase"
-	"github.com/jsuar/go-cron-descriptor/pkg/crondescriptor"
+	"github.com/iancoleman/strcase"                          // TODO factor out
+	"github.com/jsuar/go-cron-descriptor/pkg/crondescriptor" // TODO factor out
 	"github.com/pkg/errors"
 	"github.com/robfig/cron"
 	"golang.org/x/exp/slog"
 )
 
 const (
-	// TODO make MemBackend queue capacity configurable
 	defaultMemQueueCapacity = 10000 // the default capacity of individual queues
 	emptyCapacity           = 0
 )
 
 // MemBackend is a memory-backed neoq backend
 type MemBackend struct {
-	cancelFuncs      []context.CancelFunc // A collection of cancel functions to be called upon Shutdown()
-	cron             *cron.Cron
-	logger           Logger
-	jobCheckInterval time.Duration // the duration of time between checking for future jobs to schedule
-	mu               *sync.Mutex   // mutext to protect mutating state on a pgWorker
-	jobCount         int64         // number of jobs that have been queued since start
-	handlers         sync.Map      // map queue names [string] to queue handlers [Handler]
-	fingerprints     sync.Map      // map fingerprints [string] to their jobs [Job]
-	futureJobs       sync.Map      // map jobIDs [int64] to [Jobs]
-	queues           sync.Map      // map queue names [string] to queue handler channels [chan Job]
+	types.Backend
+	config       *config.Config
+	logger       logging.Logger
+	handlers     *sync.Map // map queue names [string] to queue handlers [Handler]
+	fingerprints *sync.Map // map fingerprints [string] to job [Job]
+	futureJobs   *sync.Map // map jobIDs [int64] to job [Job]
+	queues       *sync.Map // map queue names [string] to queue handler channels [chan Job]
+	cron         *cron.Cron
+	mu           *sync.Mutex          // mutext to protect mutating state on a pgWorker
+	cancelFuncs  []context.CancelFunc // A collection of cancel functions to be called upon Shutdown()
+	jobCount     int64                // number of jobs that have been queued since start
 }
 
-func NewMemBackend(opts ...ConfigOption) (n Neoq, err error) {
+// TestBackend initializes a backend for testing purposes
+func TestBackend(conf *config.Config,
```

Comment:
i would put this in `memory_backend_test.go`

Reply:
It can't go in `memory_backend_test.go` because `TestBackend` accesses private fields on the `MemBackend` struct and our tests are in the `memory_test` package, while `MemBackend` is in the `memory` package. 

I went ahead and moved `TestBackend` to `memory_backend_helper_test.go` which won't get compiled during normal builds, but is in the `memory` package, allowing it to access fields on `MemBackend`.